homeassistant/components/sensor/thermoworks_smoke.py

```diff
@@ -0,0 +1,198 @@
+
+    @property
+    def unique_id(self):
+        """Return the unique id for the sensor."""
+        return self._unique_id
+
+    @property
+    def device_info(self):
+        """Return the device info property."""
+        return {
+            'identifiers': {
+                (DOMAIN, self.serial)
+            },
+            'name': self.mgr.name(self.serial),
+            'manufacturer': THERMOWORKS_MFG,
+            'model': SMOKE_MODEL,
+            'sw_version': self.firmware_version
+        }
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._attributes
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this sensor."""
+        return self._unit_of_measurement
+
+    def update_unit(self):
+        """Set the units from the data."""
+        if PROBE_2 in self.type:
+            self._unit_of_measurement = self.mgr.units(self.serial, PROBE_2)
+        else:
+            self._unit_of_measurement = self.mgr.units(self.serial, PROBE_1)
+
+    def update(self):
+        """Get the monitored data from firebase."""
+        from stringcase import camelcase, snakecase
+        try:
+            values = self.mgr.data(self.serial)
+
+            # set state from data based on type of sensor
+            self._state = values.get(camelcase(self.type))
+
+            # set units
+            self.update_unit()
+
+            # set basic attributes for all sensors
+            self._attributes = {
+                'time': values['time'],
+                'localtime': values['localtime']
+            }
+
+            # set extended attributes for main probe sensors
+            if self.type in [PROBE_1, PROBE_2]:
+                for key, val in values.items():
+                    # add all attributes that don't contain any probe name
+                    # or contain a matching probe name
+                    if (
+                            (self.type == PROBE_1 and key.find(PROBE_2) == -1)
+                            or
+                            (self.type == PROBE_2 and key.find(PROBE_1) == -1)
+                    ):
+                        if key == BATTERY_LEVEL:
+                            key = ATTR_BATTERY_LEVEL
+                        else:
+                            # strip probe label and convert to snake_case
+                            key = snakecase(key.replace(self.type, ''))
+                        if key == FIRMWARE:
+                            self.firmware_version = val
+                        elif key:
+                            # add to attrs
+                            self._attributes[key] = val
+                # store serial number as an attribute since device_info
+                # is not visible yet
+                self._attributes['serial'] = self.serial
```

Comment:
Please don't add this.

Reply:
It should go in `device_info` if at all. So either way we don't add it in this PR.